app/src/main/cpp/src/components/transport_manager/src/bluetooth_le/bluetooth_le_device_scanner.cc

```diff
@@ -105,7 +102,32 @@ bool BluetoothLeDeviceScanner::IsInitialised() const {
 void BluetoothLeDeviceScanner::ProcessMessage(const std::vector<uint8_t>& data) {
   if (data.size() > 0) {
     SDL_LOG_DEBUG("Control message:  " << data.data());
-    // TODO: parse message from buffer
+
+    auto action = BleControlProtocol::GetMessageActionType(data);
+
+    switch(action){
+      case BleProtolActions::ON_DEVICE_CONNECTED:
+        {
+        
+          DeviceSptr device_ptr(new BluetoothLeDevice(
+              BleControlProtocol::GetAddress(data),
+              BleControlProtocol::GetName(data).c_str() ));
+
+          paired_devices_.push_back(device_ptr);
+
+          UpdateTotalDeviceList();
+        }
+      break;
+
+      case BleProtolActions::ON_DEVICE_DISCONNECTED:
+        SDL_LOG_DEBUG("Disconnect :  ");
```

Comment:
@VadymLuchko I think you should extract device address, search for it in `paired_devices_` and erase from this collection, then call `UpdateTotalDeviceList`

Reply:
fixed in 22caff3